File: app/src/main/java/me/gilo/starter/adapters/ListAdapter.java
package me.gilo.starter.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.List;

public class ListAdapter<T> extends RecyclerView.Adapter<ListViewHolder<T>> {
    private List<T> data;
    ListViewHolder<T> holder;


    public ListAdapter(List<T> data, ListViewHolder<T> holder) {
        this.data = data;
        this.holder = holder;
    }

    @Override
    public ListViewHolder<T> onCreateViewHolder(ViewGroup parent, int viewType) {
        return holder;
    }

    @Override
    public void onBindViewHolder(ListViewHolder<T> holder, int position) {
        holder.renderView(data.get(position));
    }


    @Override
    public int getItemCount() {
        return data.size() == 0 ? 0 : data.size();
    }

}

File: app/src/main/java/me/gilo/starter/adapters/ListAdapter.kt
package me.gilo.starter.adapters


import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class ListAdapter<T>(private val data: List<T>, private var holder: ListViewHolder<T>) :
    RecyclerView.Adapter<ListViewHolder<T>>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListViewHolder<T> {
        return holder
    }

    override fun onBindViewHolder(holder: ListViewHolder<T>, position: Int) {
        holder.renderView(data[position])
    }


    override fun getItemCount(): Int {
        return if (data.isEmpty()) 0 else data.size
    }

}
